### PR TITLE
Picopass block tracking

### PR DESCRIPTION
**Pull request by bettse**
# What's new

- Starting work on tracking what blocks we read.  Initially this is just about storing the value, and later iterations will add file support (`??` for unknown blocks, etc) and reading/using the value (more confidence and less inference in the UI showing the pacs data or other stuff)

# Verification 

- Picopass should run normally

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: picopass/protocol/picopass_listener.c [hunk 1]
```diff
@@ -43,11 +43,11 @@ static void picopass_listener_loclass_update_csn(PicopassListener* instance) {
     // collect LOCLASS_NUM_PER_CSN nonces in a row for each CSN
     const uint8_t* csn =
         loclass_csns[(instance->key_block_num / LOCLASS_NUM_PER_CSN) % LOCLASS_NUM_CSNS];
-    memcpy(instance->data->card_data[PICOPASS_CSN_BLOCK_INDEX].data, csn, sizeof(PicopassBlock));
+    memcpy(instance->data->card_data[PICOPASS_CSN_BLOCK_INDEX].data, csn, PICOPASS_BLOCK_LEN);
```
Comment (bettse) [inline picopass/protocol/picopass_listener.c:46]: When adding the bool to the struct I got errors because the struct size isn't the block size.

File: picopass/protocol/picopass_poller.c [hunk 2]
```diff
@@ -70,6 +70,20 @@ NfcCommand picopass_poller_select_handler(PicopassPoller* instance) {
     return command;
 }
 
+void picopass_poller_print_block(char* str, PicopassBlock block) {
```
Comment (bettse) [inline picopass/protocol/picopass_poller.c:73]: Unrelated cleanup of long and redundant lines of code

**Comment by bettse** [comment]
@nvx What do you think, as a starting point of block tracking?

**Comment by nvx** [comment]
> @nvx What do you think, as a starting point of block tracking?

Seems reasonable enough to me, although I do wonder if it'd be worth having the block array vs the is valid array separate, not sure if adding that one bool to the struct will actually result in 4 bytes of ram extra due to alignment

**Comment by bettse** [comment]
> not sure if adding that one bool to the struct will actually result in 4 bytes of ram extra due to alignment

The thought crossed my mind, but I don't think we're pushing any memory limits at this point, and I felt like this was easier to code with.  I figure if we do run into ram limits then this could more efficiently be a bitvector, but that is something for it if becomes a problem, in my opinion.

**Comment by bettse** [comment]
this is ready for review

**Review by skotopes — APPROVED** [review]
